### DISCUSSION TITLE
Graphical programming environment (like DOjS) in FreeDOS / MS-DOS in PC Emulator

### DISCUSSION THREAD

**Discussion by niutech** [fdivitto/FabGL#267] [Q&A] [answered]
Have anybody succeed in running [DOjS](https://github.com/SuperIlu/DOjS) graphical environment in FreeDOS / MS-DOS / SvarDOS running in PC Emulator? Or any other programming tool for creating graphics?

**Comment by fdivitto** [comment]
It needs at least a 80386 with 4MB

**Reply by niutech** [reply]
FabGL ESP32 Board has 4 MB of PSRAM, but is it possible to emulate 386 in protected mode?

**Reply by fdivitto** [reply]
The FabGL emulator emulates up to the 80186 instruction set.

**Reply by niutech** [reply]
Thanks for a quick reply. Do you know any other graphical programming environment which will work in FreeDOS/MS-DOS/SvarDOS on FabGL? Maybe [LoveDOS](https://github.com/rxi/lovedos), [DIV Games Studio](https://github.com/DIVGAMES/DIV-Games-Studio) or [PythonD](http://www.caddit.net/pythond/) for DOS?

**Reply by fdivitto** [reply]
None of nowadays graphical environments can run. It works with very vintage OS, like Windows 3.0, GEM, GEOS...
Maybe one day I'll add 32 bit support...

**Reply by ericjenott** [reply, marked as answer]
Visual Basic 1.0 runs fine under DOS in PCEmulator
